app/client/ui/PageWidgetPicker.ts

```diff
@@ -99,21 +99,36 @@ const testId = makeTestId('test-wselect-');
 
 // The picker disables some choices that do not make much sense. This function return the list of
 // compatible types given the tableId and whether user is creating a new page or not.
-function getCompatibleTypes(tableId: TableRef, isNewPage: boolean|undefined): IWidgetType[] {
+function getCompatibleTypes(tableId: TableRef, isNewPage: boolean|undefined, summarize: boolean): IWidgetType[] {
+  let compatibleTypes: Array<IWidgetType> = [];
   if (tableId !== 'New Table') {
-    return ['record', 'single', 'detail', 'chart', 'custom', 'custom.calendar', 'form'];
+    compatibleTypes = ['record', 'single', 'detail', 'chart', 'custom', 'custom.calendar', 'form'];
   } else if (isNewPage) {
     // New view + new table means we'll be switching to the primary view.
-    return ['record', 'form'];
+    compatibleTypes =['record', 'form'];
   } else {
     // The type 'chart' makes little sense when creating a new table.
-    return ['record', 'single', 'detail', 'form'];
+    compatibleTypes = ['record', 'single', 'detail', 'form'];
   }
+  if (summarize) {
+    return compatibleTypes.filter((el)=> !['form'].includes(el));
+  }
+  return compatibleTypes;
+}
+
+// The Picker disables some choices that do not make much sense.
+// This function return a boolean telling if summary can be used with this type.
+function isSummaryCompatible(tableId: IWidgetType): boolean{
+  const incompatibleTypes: Array<IWidgetType> = ['form'];
+  if (incompatibleTypes.indexOf(tableId) > -1){
+    return false;
+  }
+  return true;
 }
 
 // Whether table and type make for a valid selection whether the user is creating a new page or not.
-function isValidSelection(table: TableRef, type: IWidgetType, isNewPage: boolean|undefined) {
-  return table !== null && getCompatibleTypes(table, isNewPage).includes(type);
+function isValidSelection(table: TableRef, type: IWidgetType, isNewPage: boolean|undefined, summarize: boolean) {
```

Comment:
Proposal: avoid boolean flags. These can cause issues for readability on call sites.

A good practice is to rather use options (with destructuring if you want to):
```suggestion
function isValidSelection(table: TableRef, type: IWidgetType, {isNewPage, summarize}: {isNewPage: boolean|undefined, summarize: boolean}) {
```

Reply:
Thanks for the advice :)